### ISSUE TITLE
Add support for math mode in `ce()` blocks

### ISSUE THREAD

**Issue by Andrew-S-Rosen** [schang412/typst-whalogen#8]
I want to be able to do something like $\alpha + \beta \rightarrow \gamma$ in typst but currently the only way to do something like this is 

```
#import "@preview/whalogen:0.2.0": ce

$ ce("#sym.alpha + #sym.beta -> #sym.gamma") $
```

However, this renders everything as text.

In general, I think `ce()` should support entering math mode when `$` is supplied. Something like:

```
$ ce("$alpha$ + $beta$ -> $gamma$") $
#ce("$alpha$ + $beta$ -> $gamma$")
```

There are many other scenarios where it would be nice to be able to inject math, so this is really more of a general feature request.

**Comment by Andrew-S-Rosen** [comment]
Never mind. This was a "me" issue since you just need to use #sym.alpha

**Comment by schang412** [comment]
This is more difficult to implement than other requested enhancements. Currently, we immediately drop into math mode and turn all the text upright. Not sure of a good workaround for this, will require some thought. 

I’ll work on the other enhancements this weekend.

**Comment by Andrew-S-Rosen** [comment]
I agree. While I have an idea on how to do the others, this one escapes me. I figured I'd open it regardless though.

Thanks for your willingness to tackle the other ones!

**Comment by DOxidelyj** [comment]
I think I have met a similar problem: I can't use things other than numericals in subscripts & superscripts.
Is there any enhancement possible to provide this feature: for example, MX_n (MX_"n", MX_(n) or anything with a plausible syntax) will be rendered with n as the subscript?

**Comment by linuxrider** [comment]
@schang412 Would something like this make sense?
String is split at `$` and joined together subsequently.
```typst
#let test_str = "$y$ H_2O + $x^2$ ->^$k$ $k x$ N H_3"
#let str_list = test_str.split("$")

#str_list

#let str_list = if str_list.first() == "" { // eq begins with a variable, exclude empty first str, 
  str_list.slice(1,).enumerate().map(item => if calc.even(item.at(0)) {item.at(1)+" "} else {" upright("+item.at(1)+ ") "})
} else {
  str_list.enumerate().map(item => if calc.odd(item.at(0)) {item.at(1)+" "} else {" upright("+ item.at(1) + ") "})
}
#str_list

#let math_str = str_list.join().replace("upright( )", "").replace("^)", ")^").replace("upright(^", "^upright(") // correct edge cases

#math_str

#eval(math_str.trim(), mode: "math")
```

**Comment by linuxrider** [comment]
#16 covers at least very basic math mode such as rate constants.

**Comment by schang412** [comment]
Hi, this looks good. The pull request as well. Could you also some examples to the manual when you are ready for it to be merged.

**Comment by linuxrider** [comment]
Added a section about variables in the manual.

**Comment by schang412** [comment]
Hi, sorry it took so long to get around to this. Let me look test out the changes and I'll get this merged soon.
